gsmphys/sfcsub.F

```diff
@@ -2684,13 +2684,22 @@ subroutine fixrdg(lugb,idim,jdim,fngrib,
       real (kind=kind_io8) gdata(idim*jdim)
       logical gaus
       real (kind=kind_io8) blno,blto
-      real (kind=kind_io8) data8(idim*jdim)
+      real (kind=kind_io8), allocatable :: data8(:)
       real (kind=kind_io4), allocatable :: data4(:)
 !
-      logical*1 lbms(mdata)
+      logical*1, allocatable :: lbms(:)
```

Comment:
Explicit definitions, allocations, and initializations are always better than not. 

If we want to keep using the GFS Physics Driver moving forward then we may want to take the time to clean up the cruft and imprecise code within it.